### DISCUSSION TITLE
sharing the same icon image across multiple objects.

### DISCUSSION THREAD

**Discussion by keince** [pelicanmapping/osgearth#2032] [Support Q&A]
Does osgEarth have the ability to share the same icon across multiple objects? We have several hundred TrackNodes that all use the same icon image. Right now, we just load the same image each time the TrackNode is created which seems wasteful and slow.

If there's no support for this, can I use osgDB::SharedStateManager() with osgEarth to share the icon image?

Thx

Shayne

**Comment by osgwraiESC** [comment]
I was sharing the same image icon across multiple PaceNode objects (loaded once) i am pretty sure, i haven't used TrackNode but i would think the same would be true,

i had to allow each of my PlaceNode to have it's own unique color of the same image icon, so i changed to cloning the original image, then coloring it, so the worst case should be that you only have to read the image once, and clone it, 

osgEarth/ImageUtils:60:        static osg::Image* cloneImage( const osg::Image* image );

**Reply by keince** [reply]
Thank you for the response and suggestion.

I'm also changing the color of the icon but I'm doing it via a shader which is attached onto the icon geometry. I'm also rotating the icon in the same shader. 

Not sure how this will effect my attempt to share the image or texture object across all my TrackNodes...
